src/frontend.rs

```diff
@@ -100,17 +100,171 @@ impl NotificationExt for Notification {
     }
 }
 
+/// Trait for platform specific tray icons
+trait TrayIconTrait {
+    /// Initialize the tray icon
+    fn init() -> Result<GenericTrayIcon, ()>;
+    /// Set the sender to send inputs to the app
+    #[cfg(all(unix, not(target_os = "macos")))]
+    fn set_sender(&mut self, sender: AsyncComponentSender<App>);
+}
+
+/// Tray icon for Linux excluding macOS
+#[cfg(all(unix, not(target_os = "macos")))]
+struct LinuxTrayIcon {
+    icon: ksni::Icon,
+    sender: Option<AsyncComponentSender<App>>,
+}
+
+#[cfg(all(unix, not(target_os = "macos")))]
+impl ksni::Tray for LinuxTrayIcon {
+    fn id(&self) -> String {
+        env!("CARGO_PKG_NAME").to_string()
+    }
+
+    fn icon_name(&self) -> String {
+        "Space Acres".to_string()
+    }
+
+    fn title(&self) -> String {
+        "Space Acres".to_string()
+    }
+
+    fn icon_pixmap(&self) -> Vec<ksni::Icon> {
+        vec![self.icon.clone()]
```

Comment:
I think it is supposed to be constructed on demand, I'd probably check docs and usage and not store the icon in RAM all the time.